packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -210,30 +213,37 @@ class VisualStudio {
     _cachedUsableVisualStudioDetails ??= _visualStudioDetails(
         requiredComponents: _requiredComponents().keys,
         additionalArguments: <String>[_prereleaseKey]);
-    if (_cachedUsableVisualStudioDetails != null) {
-      if (installationHasIssues(_cachedUsableVisualStudioDetails)) {
+    // If an installation with required components if found, but it has issues (e.g. is not complete)
+    // or none is found, assign a sentinel empty value to avoid querying vswhere again.
+    if (_cachedUsableVisualStudioDetails != null && _cachedUsableVisualStudioDetails.isNotEmpty) {
+      if (installationHasIssues(_cachedUsableVisualStudioDetails)) {  
         _cachedAnyVisualStudioDetails = _cachedUsableVisualStudioDetails;
-        return null;
+        _cachedUsableVisualStudioDetails = <String, dynamic>{};
```

Comment:
Optional: I found this re-assignment construction (where the usable version variable is set to a value that might not actually be usable, then unset if needed) a bit hard to follow; I think a local variable for the lookup would make it clearer:
```
if (_cachedUsableVisualStudioDetails == null) {
  Map<String, dynamic> visualStudioWithComponentsDetails = [...];
  visualStudioWithComponentsDetails ??= [...];
  if (visualStudioWithComponentsDetails != null) {
    if (installationHasIssues(visualStudioWithComponentsDetails) {
      _cachedUsable... = visualStudioWithCompontesDetails;
    } else {
      _cachedAny... = visualStudioWithCompontesDetails;
    }
  }
  _cachedUsableVisualStudioDetails ??= <String, dynamic>{};
}
return _cachedUsableVisualStudioDetails;
```